app/models/shipit/stack.rb

```diff
@@ -642,9 +642,7 @@ def schedule_merges_if_necessary
     end
 
     def emit_lock_hooks
-      return unless previous_changes.include?('lock_reason')
-
-      lock_details = if previous_changes['lock_reason'].last.blank?
+      lock_details = if previous_changes.include?('lock_reason') && previous_changes['lock_reason'].last.blank?
```

Comment:
Am I understanding this right, that the issue is that with this rails bug, `previous_changes` doesn't include the lock information?

In that case isn't this now just sending blank webhooks (at least as far as locking info)? I feel like that isn't going to be any better than before and might break stuff if it's expecting these fields to be set?

I don't really have a better solution. I'm not opposed to an `after_save` as the transaction can only roll back the DB and any third party api calls are still going to happen right away, so it's not exactly perfect the way it is. 

Maybe there's a way to separate the two so that the rollback isn't a side effect of aborting and instead we do them in succession? That would make them separate transactions. The webhooks are working for regular rollbacks

Reply:
We can't do it from `after_change`, if we did the ob could be ran before the transaction is committed, and send the wrong data.

Reply:
Both of what you said makes sense - yeah, I think making them separate transactions is a cleaner fix but a riskier one. And yes, the issue is that `previous_changes` doesn't include the lock info in the case of "Abort & Rollback." I'm going to create a new PR that just calls `emit_lock_hooks` directly from within `trigger_revert`. I'll have that up in a moment and will close this - thank you both for the feedback!